cpp/arcticdb/pipeline/frame_utils.hpp

```diff
@@ -142,12 +142,21 @@ void aggregator_set_data(
                     } else if(is_py_nan(*ptr_data)){
                         agg.set_no_string_at(col, s, nan_placeholder());
                     } else {
+                        std::variant<convert::StringEncodingError, convert::PyStringWrapper> wrapper_or_error;
                         if constexpr (is_utf_type(slice_value_type(dt))) {
-                            auto wrapper= convert::py_unicode_to_buffer(*ptr_data, scoped_gil_lock);
-                            agg.set_string_at(col, s, wrapper.buffer_, wrapper.length_);
+                            wrapper_or_error = convert::py_unicode_to_buffer(*ptr_data, scoped_gil_lock);
                         } else {
-                            auto wrapper = convert::pystring_to_buffer(*ptr_data, false);
-                            agg.set_string_at(col, s, wrapper.buffer_, wrapper.length_);
+                            wrapper_or_error = convert::pystring_to_buffer(*ptr_data, false);
+                        }
+                        // Cannot use util::variant_match as only one of the branches would have a return type
```

Comment:
You can use variant matching if you specify the return type I think, as in `[] (const auto&) -> std::optional<convert::StringEncodingError> ( util:raise("blah")); }`

Reply:
I don't see how variant match will improve things. We have different cases based on the contained type, there is no duck-typed branch here allowing to do the same thing for a few types.

Reply:
What I mean is that we only return from `aggregator_set_data` if the variant holds `StringEncodingError`, otherwise we continue through the loop.

Reply:
Fair enough, I guess in the case where there's no common return type this way is probably better.